src/main/java/org/lwjglx/util/glu/MipMap.java

```diff
@@ -64,57 +65,59 @@ public static int gluBuild2DMipmaps(final int target, final int components, fina
         int retVal = 0;
         boolean done = false;
 
-        if (w != width || h != height) {
-            // must rescale image to get "top" mipmap texture image
-            image = BufferUtils.createByteBuffer((w + 4) * h * bpp);
-            int error = gluScaleImage(format, width, height, type, data, w, h, type, image);
-            if (error != 0) {
-                retVal = error;
-                done = true;
-            }
+        try (MemoryStack stack = stackPush()) {
+            if (w != width || h != height) {
+                // must rescale image to get "top" mipmap texture image
+                image = stack.malloc((w + 4) * h * bpp);
```

Comment:
This cannot be stack-allocated, the full atlas image can get too big for the memorystack size